### PR TITLE
Add flag sbt.gigahorse.enabled, to optionally disable Gigahorse

### PR DESCRIPTION
**Pull request by cunei**
The default for the flag is true.

### CODE REVIEW

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?librarymanagement=pull/167/head&deploy_to_bintray=false&run_sbt_community=false">this link.</A>

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/175/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: 1.x-1.x-1.x-pull/167/head-1.x-1.x)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/175/">has checked</A> the following projects against Scala 2.12,
tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/tree/1.x">1.x                 </A> | sbt/sbt@34945af6314dee3c9ea55a76c882007f44819509
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/1.x">1.x                 </A> | sbt/zinc@ef3a2c48a4e6b6126c8f1d0a99e530d64925c7d3
  io                | <A HREF="https://github.com/sbt/io/tree/1.x">1.x                 </A> | sbt/io@62004b28dfcdc33273ebb07a563e78178f6eb21d
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/pull/167">pull/167/head       </A> | sbt/librarymanagement@b480c9c65b2220443b030823208ca3036df69b96
  util              | <A HREF="https://github.com/sbt/util/tree/1.x">1.x                 </A> | sbt/util@cba6f83447b84cc1e967f25a930d4f411f6f0664
  website           | <A HREF="https://github.com/sbt/website/tree/1.x">1.x                 </A> | 

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=1.x&zinc=1.x&io=1.x&librarymanagement=pull/167/head&util=1.x&website=1.x&deploy_to_bintray=false">(restart)</A>

**Comment by typesafe-tools** [comment]
A validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt-validator/176/console">is in progress...</a>

**Comment by typesafe-tools** [comment]
[//]: # (Refs: 1.x-1.x-1.x-pull/167/head-1.x-1.x)
The validator <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/176/">has checked</A> the following projects against Scala 2.12,
tested using dbuild, projects built on top of each other.

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/tree/1.x">1.x                 </A> | sbt/sbt@34945af6314dee3c9ea55a76c882007f44819509
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/1.x">1.x                 </A> | sbt/zinc@ef3a2c48a4e6b6126c8f1d0a99e530d64925c7d3
  io                | <A HREF="https://github.com/sbt/io/tree/1.x">1.x                 </A> | sbt/io@62004b28dfcdc33273ebb07a563e78178f6eb21d
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/pull/167">pull/167/head       </A> | sbt/librarymanagement@b480c9c65b2220443b030823208ca3036df69b96
  util              | <A HREF="https://github.com/sbt/util/tree/1.x">1.x                 </A> | sbt/util@cba6f83447b84cc1e967f25a930d4f411f6f0664
  website           | <A HREF="https://github.com/sbt/website/tree/1.x">1.x                 </A> | 

:white_check_mark: The result is: ***SUCCESS***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild?sbt=1.x&zinc=1.x&io=1.x&librarymanagement=pull/167/head&util=1.x&website=1.x&deploy_to_bintray=false">(restart)</A>

**Comment by cunei** [comment]
sbt community build works ok when using this flag: https://jenkins.scala-sbt.org:8592/job/sbt-community/26/

File: ivy/src/main/scala/sbt/librarymanagement/ivy/UpdateOptions.scala [hunk 1]
```diff
@@ -109,7 +109,7 @@ object UpdateOptions {
       interProjectFirst = true,
       latestSnapshots = true,
       cachedResolution = false,
-      gigahorse = true,
+      gigahorse = sys.props.get("sbt.gigahorse.enabled") map { _.toLowerCase == "true" } getOrElse true,
```
Comment (eed3si9n) [inline ivy/src/main/scala/sbt/librarymanagement/ivy/UpdateOptions.scala]: I think "enabled" is redundant. `sbt.gigahorse`?

**Review by eed3si9n — CHANGES_REQUESTED** [review]
Nitpicking on the word.

File: ivy/src/main/scala/sbt/librarymanagement/ivy/UpdateOptions.scala (hunk 1 above)
Comment (cunei) [reply ivy/src/main/scala/sbt/librarymanagement/ivy/UpdateOptions.scala]: I considered that there might be further gigahorse-related flags in the future. Hence `gigahorse.enabled`, `gigahorse.xxx`, `gigahorse.yyy`,... etc
Besides, in general boolean flags are clearer when named as a predicate: `sbt.gigahorse` by itself does not make it apparent what the value represents.

Comment (eed3si9n) [reply ivy/src/main/scala/sbt/librarymanagement/ivy/UpdateOptions.scala]: Not really likely, and even if it happens I think it's fine to have `sbt.gigahorse` and `sbt.gigahorse.followredirect` ect.

**Comment by typesafe-tools** [comment]
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?librarymanagement=pull/167/head&deploy_to_bintray=false&run_sbt_community=false">this link.</A>

**Review by eed3si9n — APPROVED** [review]

**Comment by eed3si9n** [comment]
Ref https://github.com/sbt/sbt/issues/3519

**Comment by swsnr** [comment]
Also helps with https://github.com/sbt/sbt/issues/3585.

**Pull request merged**